libs/common/AsciiUtils.cs

```diff
@@ -1,17 +1,83 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System;
+using System.Diagnostics;
+using System.Text;
+
 namespace Garnet.common;
 
 /// <summary>
 /// Utilites for ASCII parsing and manipulation.
 /// </summary>
+/// <remarks>
+/// This class polyfills various <see cref="char"/> and <c>Ascii</c> methods for .NET 6.
+/// </remarks>
 public static class AsciiUtils
 {
-    public static byte ToLower(byte value)
+    /// <summary>Indicates whether a character is within the specified inclusive range.</summary>
+    /// <param name="c">The character to evaluate.</param>
+    /// <param name="minInclusive">The lower bound, inclusive.</param>
+    /// <param name="maxInclusive">The upper bound, inclusive.</param>
+    /// <returns>true if <paramref name="c"/> is within the specified range; otherwise, false.</returns>
+    /// <remarks>
+    /// The method does not validate that <paramref name="maxInclusive"/> is greater than or equal
+    /// to <paramref name="minInclusive"/>.  If <paramref name="maxInclusive"/> is less than
+    /// <paramref name="minInclusive"/>, the behavior is undefined.
+    /// </remarks>
+    public static bool IsBetween(byte c, char minInclusive, char maxInclusive)
+    {
+        return (uint)(c - minInclusive) <= (uint)(maxInclusive - minInclusive);
+    }
+
+    public static byte ToLower(byte c)
+    {
+        if (IsBetween(c, 'A', 'Z'))
+            c = (byte)(c | 0x20);
+        return c;
+    }
+    public static byte ToUpper(byte c)
+    {
+        if (IsBetween(c, 'a', 'z'))
+            c = (byte)(c & ~0x20);
+        return c;
+    }
+
+    /// <summary>
+    /// Convert ASCII Span to upper case
+    /// </summary>
+    public static void ToUpperInPlace(Span<byte> command)
     {
-        if ((uint)(value - 'A') <= (uint)('Z' - 'A')) // Is in [A-Z]
-            value = (byte)(value | 0x20);
-        return value;
+#if NET8_0_OR_GREATER
+        Ascii.ToUpperInPlace(command, out _);
+#else
+        foreach (ref var c in command)
+            if (c > 96 && c < 123)
+                c -= 32;
+#endif
+    }
+
+    /// <summary>
+    /// Check if two byte spans are equal, where right is an all-upper-case span, ignoring case if there are ASCII bytes.
+    /// </summary>
+    public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
```

Comment:
Can we rename this back to `EqualsUpperCaseSpanIgnoringCase`? Else the consumer may use it in wrong situation where `right` is not known to be uppercase.

Reply:
We had it as an extension method earlier, so the name name more sense. `A.EqualsUpperCaseSpanIgnoringCase(B)`.

Reply:
> Why did you change it from extension method to normal.

Mostly consistency with other `*Utils` & allowing PR reviewers to know exact method which a line is calling by glance & without having to hover in IDE. 

Reverted to original shape.